cmd/panfs.go

```diff
@@ -519,29 +502,17 @@ func (fs *PANFSObjects) MakeBucketWithLocation(ctx context.Context, bucket strin
 
 	defer NSUpdated(bucket, slashSeparator)
 
-	bucketDir, err := fs.getBucketDir(ctx, bucket)
-	if err != nil {
-		return toObjectErr(err, bucket)
-	}
-	// fsMkdir is fs-v1 specific method. Maybe it makes sense to create panfs-helpers.go...
-	if err = fsMkdir(ctx, bucketDir); err != nil {
-		return toObjectErr(err, bucket)
-	}
-
 	bucketMetaDir := pathJoin(opts.PanFSBucketPath, bucket, panfsMetaDir)
-	// Create dir for object metadata
-	if err := mkdirAll(pathJoin(bucketMetaDir, objMetadataDir), 0o777); err != nil {
-		return toObjectErr(err, bucket)
-	}
-
-	// Create dir for temporary uploads
-	if err := mkdirAll(pathJoin(bucketMetaDir, tmpDir), 0o777); err != nil {
-		return toObjectErr(err, bucket)
-	}
-
-	// Create dir for multipart uploads
-	if err := mkdirAll(pathJoin(bucketMetaDir, mpartMetaPrefix), 0o777); err != nil {
-		return toObjectErr(err, bucket)
+	for _, dir := range []string{
+		pathJoin(opts.PanFSBucketPath, bucket),
+		bucketMetaDir,
+		pathJoin(bucketMetaDir, objMetadataDir),
+		pathJoin(bucketMetaDir, tmpDir),
+		pathJoin(bucketMetaDir, mpartMetaPrefix),
+	} {
+		if err := fsMkdir(ctx, dir); err != nil {
+			return toObjectErr(err, bucket)
+		}
```

Comment:
Use the os.MkdirAll function instead of fsMkdir.
If the directories are hierarchical and do not exist, you can use the os.MkdirAll function to create them in one call instead of calling fsMkdir multiple times. Here is an example implementation:
`dirs := []string{
    pathJoin(opts.PanFSBucketPath, bucket),
    bucketMetaDir,
    pathJoin(bucketMetaDir, objMetadataDir),
    pathJoin(bucketMetaDir, tmpDir),
    pathJoin(bucketMetaDir, mpartMetaPrefix),
}
dirPath := filepath.Join(dirs...)
if err := os.MkdirAll(dirPath, 0755); err != nil {
    return toObjectErr(err, bucket)
}
`

Reply:
> Use the os.MkdirAll function instead of fsMkdir. If the directories are hierarchical and do not exist, you can use the os.MkdirAll function to create them in one call instead of calling fsMkdir multiple times. Here is an example implementation: `dirs := []string{ pathJoin(opts.PanFSBucketPath, bucket), bucketMetaDir, pathJoin(bucketMetaDir, objMetadataDir), pathJoin(bucketMetaDir, tmpDir), pathJoin(bucketMetaDir, mpartMetaPrefix), } dirPath := filepath.Join(dirs...) if err := os.MkdirAll(dirPath, 0755); err != nil { return toObjectErr(err, bucket) }`

Here I need to create the following structure of directories for each new bucket:
```
<panfs_bucket_path>/<bucket_name>
    `-- .s3
        |-- metadata
        |-- multipart
        `-- tmp
```
 minio provided with its own implementation of `mkdirAll` but the idea here is to do not allow to create a bucket if parent directory does not exists so using of `mkdirAll` is not convenient here.